scripts/shell.sh

```diff
@@ -205,15 +221,60 @@ ensure_nix_is_present() {
   setup_nix
 }
 
+ensure_vanilla_rc_exists() {
+  mkdir -p "${CACHE_ROOT}"
+  cat > "${VANILLA_RC}" << EOF
+. ${USER_HOME}/.nix-profile/etc/profile.d/nix.sh
+export NIX_CONF_DIR=${NIX_CONF_DIR}
+nix-channel --add https://nixos.org/channels/nixos-21.11 nixos
+nix-channel --update
+export NIX_PATH=${USER_HOME}/.nix-defexpr/channels:nixpkgs=/nix/var/nix/profiles/per-user/${USER_NAME}/channels/nixos
```

Comment:
I'd suggest using something like:
`export NIX_PATH=nixpkgs=https://github.com/NixOS/nixpkgs/archive/refs/heads/nixos-21.11.tar.gz`

This way you won't modify nix channels outside of nix-shell

Reply:
Or even better, expose it as a command line argument.